test/compilable/fix17349.d

```diff
@@ -0,0 +1,25 @@
+/* TEST_ARGS:
+ * PERMUTE_ARGS:
+ */
+
+// https://issues.dlang.org/show_bug.cgi?id=17349
+
+struct S { }
+
+class C {
+    void bar();
+    void foo(void* p);
+    void abc(Object);
+    void def(S);
+}
+
+class D : C {
+    override void bar() const;
+    override void foo(const void*);
+    override void abc(const Object);
+    override void def(const S);
+}
+
+alias fp_t = void function(int*);
+void abc(const int*);
+fp_t fp = &abc;
```

Comment:
Doesn't that fix https://issues.dlang.org/show_bug.cgi?id=16303 also ?
Could you add a test case with delegates ?